pkg/ext-proc/main.go

```diff
@@ -221,6 +237,30 @@ func startExternalProcessorServer(
 	return svr
 }
 
+func startMetricsHandler(port int) *http.Server {
+	metrics.Register()
+
+	var svr *http.Server
+	go func() {
+		klog.Info("Starting metrics HTTP handler ...")
+
+		mux := http.NewServeMux()
+		mux.Handle("/metrics", promhttp.HandlerFor(
+			legacyregistry.DefaultGatherer,
+			promhttp.HandlerOpts{},
+		))
+
+		svr = &http.Server{
+			Addr:    net.JoinHostPort("", strconv.Itoa(port)),
+			Handler: mux,
+		}
+		if err := svr.ListenAndServe(); err != http.ErrServerClosed {
+			klog.Fatalf("failed to start metrics HTTP handler: %v", err)
+		}
```

Comment:
we need to have this rbac protected so that only authorized users on the cluster can scrape the metrics, right?

Reply:
See this example: https://github.com/kubernetes-sigs/lws/blob/b23f068de44d81a06f43a5a4fbe32993a19294ae/cmd/main.go#L123

Reply:
Added auth support.